src/clarity-angular/emphasis/alert/providers/icon-and-types-service.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class AlertIconAndTypesService {
+    private defaultIconShape = "info-circle";
+    private _alertIconShape = "";
+    private _alertType = "info";
+
+    get alertType(): string {
+        return this._alertType;
+    }
+    set alertType(val: string) {
+        if (this.alertTypes.indexOf(val) > -1) {
+            this._alertType = val;
+        }
+    }
+
+    // alert-* types are deprecated!
+    private alertTypes: string[] = [
+        "alert-info", "alert-warning", "alert-danger", "alert-success",
+        "info", "warning", "danger", "success"
+    ];
+
+    get alertIconShape(): string {
+        if ("" === this._alertIconShape) {
+            return this.iconInfoFromType(this._alertType);
+        }
+        return this._alertIconShape;
+    };
+    set alertIconShape(val: string) {
+        if (!val) {
+            this._alertIconShape = "";
+        } else if (val !== this._alertIconShape) {
+            this._alertIconShape = val;
+        }
+    }
+
+    public iconInfoFromType(type: string, classOrShape: string = "shape"): string {
+        let returnShape = "";
+        let returnClass = "";
+
+        switch (type) {
+            case "warning":
+            case "alert-warning":
+                returnShape = "exclamation-triangle";
+                returnClass = "alert-warning";
+                break;
+            case "danger":
+            case "alert-danger":
+                returnShape = "exclamation-circle";
+                returnClass = "alert-danger";
+                break;
+            case "success":
+            case "alert-success":
+                returnShape = "check-circle";
+                returnClass = "alert-success";
+                break;
+            default:
+                returnShape = this.defaultIconShape;
+                returnClass = "alert-info";
+                break;
+        }
+
+        return "shape" === classOrShape ? returnShape : returnClass;
```

Comment:
I'd feel more comfortable if you simply returned
```typescript
{
  shape: returnShape,
  cssClass: returnClass
}
```
Relying on a 2-value string as an argument is a hack, in particular it can't be typechecked. Just having callers do:
```typescript
service.iconInfoFromType("warning").shape
```
is strictly less characters (one space and 2 quotes less, to be precise :trollface: ) and throws errors at compile time as opposed to runtime.